File: tests/e2e/status_test.go
package e2e

import (
	"strings"
	"testing"
)

// After a sync with no source changes, every enabled vendor must report "in
// sync" against the would-be generation.
func TestStatus_InSyncAfterSync(t *testing.T) {
	if isWindows() {
		t.Skip("symlink tests require Unix")
	}
	h := newHarness(t)
	h.bootstrap()
	h.mustRun("sysprompt", "set", "stable")
	h.mustRun("sync")

	stdout, _ := h.mustRun("status")
	if !strings.Contains(stdout, "Would-be generation:") {
		t.Errorf("expected would-be generation line, got:\n%s", stdout)
	}
	if !strings.Contains(stdout, "in sync") {
		t.Errorf("expected at least one vendor 'in sync', got:\n%s", stdout)
	}
}

// Changing the prompt after a sync must surface as drift in status, since the
// would-be generation no longer matches what is activated.
func TestStatus_DriftAfterSourceChange(t *testing.T) {
	if isWindows() {
		t.Skip("symlink tests require Unix")
	}
	h := newHarness(t)
	h.bootstrap()
	h.mustRun("sysprompt", "set", "v1")
	h.mustRun("sync")

	// Change the source but do NOT sync.
	h.mustRun("sysprompt", "set", "v2-unsynced")

	stdout, _ := h.mustRun("status")
	if !strings.Contains(stdout, "drifted") {
		t.Errorf("expected drift after unsynced source change, got:\n%s", stdout)
	}
}

// A disabled vendor must be reported as "disabled" rather than drifted.
func TestStatus_DisabledVendorReported(t *testing.T) {
	if isWindows() {
		t.Skip("symlink tests require Unix")
	}
	h := newHarness(t)
	h.bootstrap()

	cfg := h.readFile(h.configFile())
	cfg = strings.Replace(cfg,
		"[vendors.codex]\n    enabled = true",
		"[vendors.codex]\n    enabled = false", 1)
	h.writeFile(h.configFile(), cfg)

	stdout, _ := h.mustRun("status")
	if !strings.Contains(stdout, "disabled") {
		t.Errorf("expected a 'disabled' vendor in status, got:\n%s", stdout)
	}
}

File: apps/ponte/tests/e2e/status.test.ts
import { describe, expect, it } from "bun:test";
import { newHarness, skillEntry } from "./harness";

describe("status", () => {
  it("reports in sync after a sync", async () => {
    const h = await newHarness();
    await h.bootstrap();
    await h.mustRun("sysprompt", "set", "stable");
    await h.mustRun("sync");

    const { stdout } = await h.mustRun("status");
    expect(stdout).toContain("System prompt: AGENTS.md");
    expect(stdout).toContain("in sync");
    await h.close();
  });

  it("reports drift when a link the config dropped is still present", async () => {
    const h = await newHarness();
    await h.bootstrap();
    await h.mustRun("sysprompt", "set", "v1");

    const cfg = await h.readFileText(h.configPath("config.toml"));
    await h.appendConfig(skillEntry(h.fixtureDir("simple_skill")));
    await h.mustRun("sync");
    await h.writeFile(h.configPath("config.toml"), cfg);

    const { stdout } = await h.mustRun("status");
    expect(stdout).toContain("drifted");
    await h.close();
  });

  it("reports a disabled vendor as disabled", async () => {
    const h = await newHarness();
    await h.bootstrap();

    const cfg = await h.readFileText(h.configPath("config.toml"));
    const withDisabled = cfg.replace(
      "[vendors.codex]\nenabled = true",
      "[vendors.codex]\nenabled = false",
    );
    await h.writeFile(h.configPath("config.toml"), withDisabled);

    const { stdout } = await h.mustRun("status");
    expect(stdout).toContain("disabled");
    await h.close();
  });
});
